curvefs/src/client/fuse_s3_client.cpp

```diff
@@ -64,92 +64,22 @@ void FuseS3Client::UnInit() {
     FuseClient::UnInit();
 }
 
-void FuseS3Client::FuseOpInit(void *userdata, struct fuse_conn_info *conn) {
+CURVEFS_ERROR FuseS3Client::CreateFs(
+    void *userdata, FsInfo *fsInfo) {
     struct MountOption *mOpts = (struct MountOption *) userdata;
-    std::string mountPointStr =
-        (mOpts->mountPoint == nullptr) ? "" : mOpts->mountPoint;
     std::string fsName = (mOpts->fsName == nullptr) ? "" : mOpts->fsName;
-    std::string user = (mOpts->user == nullptr) ? "" : mOpts->user;
-
-    std::string mountPointWithHost;
-    int retVal = AddHostNameToMountPointStr(mountPointStr, &mountPointWithHost);
-    if (retVal < 0) {
-        return;
-    }
-
-    FsInfo fsInfo;
-    FSStatusCode ret = mdsClient_->GetFsInfo(fsName, &fsInfo);
-    if (ret != FSStatusCode::OK) {
-        if (FSStatusCode::NOT_FOUND == ret) {
-            LOG(INFO) << "The fsName not exist, try to CreateFs"
-                      << ", fsName = " << fsName;
-
-            ::curvefs::common::S3Info s3Info;
-            s3Info.set_ak(option_.s3Opt.s3AdaptrOpt.ak);
-            s3Info.set_sk(option_.s3Opt.s3AdaptrOpt.sk);
-            s3Info.set_endpoint(option_.s3Opt.s3AdaptrOpt.s3Address);
-            s3Info.set_bucketname(option_.s3Opt.s3AdaptrOpt.bucketName);
-            s3Info.set_blocksize(option_.s3Opt.blocksize);
-            s3Info.set_chunksize(option_.s3Opt.chunksize);
-            // TODO(xuchaojie) : where to get 4096?
```

Comment:
TODO is been removed, but you didn't solve the problem

Reply:
fsBlockSize means min allocsize, for s3, we do not need this